File: Challenge 9.py
'''
Write a program that will generate a
random playing card e.g. ‘9 Hearts’,
‘Queen Spades’ when the Enter key is
pressed.
Rather than generate a random number
from 1 to 52. Create two random numbers – one for the suit and one for the
card.

Extension
Make a loop structure so playing cards can keep being generated 
'''

import random
import keyboard

types = ['Clubs', 'Diamonds', 'Hearts', 'Spades']
ranks = ['Ace', '2', '3', '4', '5', '6', '7', '8', '9', '10', 'Jack', 'Queen', 'King']



print('Press Enter to pick a card.')
while True:
    keyboard.wait('Enter')
    x = random.randint(0, 3)
    y = random.randint(0, 12)
    card = ranks[y] + ' ' + types[x]
    print(card)

File: Challenge 9.java
import java.util.Random;
import java.util.Scanner;

class CardPicker {
    static String[] types = {"Clubs", "Diamonds", "Hearts", "Spades"};
    static String[] ranks = {"Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King"};

    public static String getRandomElement(String[] array) {
        Random r = new Random();
        int random_index = r.nextInt(array.length);
        return array[random_index];
    }

    public static String getInput() {
        System.out.print(">> ");
        Scanner sc_input = new Scanner(System.in);
        String input = sc_input.nextLine();
        return input;
    }

    public static void main(String[] args) {
        boolean run = true;
        System.out.println("Press any key to pick a card, Exit to exit.");
        while (run == true) {
            String type = getRandomElement(types);
            String rank = getRandomElement(ranks);
            String result = rank + " " + type;
            String input = getInput();
            if (input.toLowerCase().equals("exit")) {
                run = false;
            } else {
                System.out.println(result);                
            }
        }
    }
}
